Sleepy ES5503 docs (#29)
LTVA1/furnace: doc/4-instrument/es5503.md

# ES5503 instrument editor

> ES5503 instrument editor is divided into 4 tabs:

Reviewer: ```suggestion
the ES5503 instrument editor is divided into 4 tabs:
```

- **ES5503**: for controlling the parameters of ES5503 sound channel.
- **Wavetable**: for wavetable-related controls.
- **Macros**: for macros controlling volume/arp/pitch/panning/etc.
- **Sample**: for sample-related controls.

## ES5503

> There is a tricky feature of ES5503 when it comes to playing wavetables/samples. ES5503 can play waves of sizes 256, 512, ..., 32768 steps. When it encounters `0` in sample memory it stops playing the wavetable/sample, no matter what oscillator mode and wave size are.

Reviewer: Perhaps "waves of sizes that are multiples of 256"? Either that or add "768" to the sequence to make it clear.

Author: I should have continued... It's 256, 512, 1024, 2048, 4096, 8192, 16384 and 32768

- **Oscillator mode**: lets you select one of four possible oscillator modes:
  - **Freerun**: oscillator plays the whole wavetable/sample and loops back if it haven't encountered `0` in sample memory along the way. So if sample does not exactly fit to the one of aforementioned possible sizes it will stop playing after being played once.
  - **Oneshot**: oscillator plays the whole wavetable/sample once.

>   - **Sync/AM**: sync mode is only possible between pairs of even and odd numbered oscillators, i.e. a lower even-numbered oscillator paired with a higher odd-numbered oscillator. When the even-numbered oscillator starts playing back its wavetable, its odd-numbered mate will be synchronized to it and begins its wavetable/sample simultaneously (aka hard sync). (shamelessly stolen from [here](http://www.buchty.net/ensoniq/5503.html)) AM means amplitude modulation and works in reverse - odd-numbered oscillator modulates the volume of previous even-numbered oscillator. It's kinda tricky to work with this mode.

Reviewer: ```suggestion
  - **Sync/AM**: sync mode is only possible between pairs of even and odd numbered oscillators, i.e. a lower even-numbered oscillator paired with a higher odd-numbered oscillator. When the even-numbered oscillator starts playing back its wavetable, its odd-numbered mate will be synchronized to it and begins its wavetable/sample simultaneously, aka hard sync. (info from [www.buchty.net](http://www.buchty.net/ensoniq/5503.html) AM means amplitude modulation and works in reverse; odd-numbered oscillator modulates the volume of previous even-numbered oscillator. It's kinda tricky to work with this mode.
```

I don't want to strip the personality from everything, but "shamelessly stolen" might be a bit much. :)

  - **Swap**: uses even/odd pairs of oscillators as explained above. Enabled oscillator plays its wavetable/sample once, then stops as if in oneshot mode, and launches its mate. This is how it is possible to play 64 KiB sample seamlessly.
- **Virtual softpan channel**: this is a software feature. Under the hood is just uses a pair of odd and even numbered channels, assigns them with the same wave, oscillator mode and frequency. Thus you effectively have one channel with 256-step panning. Furnace provides macros for individual left and right panning control.

> - **Phase reset on key-on**: resets phase of the channel each time new note plays. Does not apply when `03xx` and `EAxx` effects are used on the same row.

Reviewer: ```suggestion
- **Phase reset on key-on**: resets phase of the channel each time a new note plays. Does not apply when `03xx` and `EAxx` effects are used on the same row.
```
